File: easy-go-app/backend/ent/schema/task.go
package schema

import (
	"time"

	"entgo.io/ent"
	"entgo.io/ent/schema/field"
	"github.com/google/uuid"
)

// Task holds the schema definition for the Task entity.
type Task struct {
	ent.Schema
}

// Fields of the Task.
func (Task) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.New()).Default(uuid.New),
		field.String("title").NotEmpty(),
		field.String("description").NotEmpty(),
		field.Time("created_at").Default(time.Now),
		field.Time("updated_at").Default(time.Now).UpdateDefault(time.Now),
		field.Enum("status").Values("todo", "in_progress", "done").Default("todo"),
	}
}

// Edges of the Task.
func (Task) Edges() []ent.Edge {
	return nil
}

File: easy-go-app/frontend/src/types/task.ts
/**
 * // Task is the model entity for the Task schema.
 * type Task struct {
 * 	config `json:"-"`
 * 	// ID of the ent.
 * 	ID uuid.UUID `json:"id,omitempty"`
 * 	// Title holds the value of the "title" field.
 * 	Title string `json:"title,omitempty"`
 * 	// Description holds the value of the "description" field.
 * 	Description string `json:"description,omitempty"`
 * 	// CreatedAt holds the value of the "created_at" field.
 * 	CreatedAt time.Time `json:"created_at,omitempty"`
 * 	// UpdatedAt holds the value of the "updated_at" field.
 * 	UpdatedAt time.Time `json:"updated_at,omitempty"`
 * 	// Status holds the value of the "status" field.
 * 	Status task.Status `json:"status,omitempty"`
 * }
 */
export enum EStatus {

}

export interface Task {
	id: string;
	title: string;
	description: string;
	created_at: Date;
	updated_at: Date;
};